src/main/java/dev/esophose/playerparticles/manager/ParticleManager.java

```diff
@@ -195,6 +194,11 @@ public void run() {
         // Spawn particles for each player
         for (PPlayer pplayer : this.particlePlayers.values()) {
             Player player = pplayer.getPlayer();
+            
+            if(player != null) {
```

Comment:
This if-section needs to occur in a prior loop to trying to send them the particles, otherwise it's updating late.

Example iteration:

Player A, set participating and afk
displayParticles(player A, ...)
  -> iterates thru Player A, Player B, Player C. B and C did not have it set
  
Player B, set participating and afk
displayParticles(player B, ...)
  -> iterates thru Player A, Player B, Player C. C did not have it set
  
Player C, set participating and afk
displayParticles(player C, ...)
  -> iterates thru Player A, Player B, Player C. now all of them were correct